mock/environment/__init__.py

```diff
@@ -167,6 +167,39 @@ def __init__(self):
                 logging.info(f'Writing {output_filename}')
                 pd.to_pickle(self.site_data_by_night[site], output_filename)
 
+
+    def get_weather(self, site: Site, start_time: datetime, end_time: datetime) -> list[object]:
+        """
+        Returns list of weather data
+        based off start and end times 
+        """
+        weather_list = []
+
+        def round_seconds(obj: datetime) -> datetime:
+            if obj.microsecond >= 500_000:
+                obj += timedelta(seconds=1)
+            return obj.replace(microsecond=0)
+
+        if start_time.date() in self.site_data_by_night[site] and end_time.date() in self.site_data_by_night[site]:
+            start_time = round_seconds(start_time)
+            end_time = round_seconds(end_time)
+
+            if start_time <= end_time:
+                if start_time in set(self.site_data_by_night[site][start_time.date()][Env._time_stamp]) and end_time in set(self.site_data_by_night[site][end_time.date()][Env._time_stamp]):
+                    for night in self.site_data_by_night[site]:
+                        for index, data in self.site_data_by_night[site][night].iterrows():
+                            if data[Env._time_stamp] >= start_time and data[Env._time_stamp] <= end_time:        
```

Comment:
This is fine but like I mentioned up there a few comments ago, you can make it a little easier to read by writing it as:
```python
if start_time <= data[Env._time_stamp] <= end_time:
```